### ISSUE TITLE
DynamoDB enhanced client index.query() returns SdkIterable instead of PageIterable

### ISSUE THREAD

**Issue by superwese** [aws/aws-sdk-java-v2#2185]
## Describe the issue
The README states that querying an index returns a PageIterable.
```
DynamoDbIndex<Customer> customersByName = customerTable.index("customers_by_name");    
PageIterable<Customer> customersWithName = 
    customersByName.query(r -> r.queryConditional(**keyEqualTo**(k -> k.partitionValue("Smith"))));
```
When I try to do this, my IDE complains
```
Required type: PageIterable<Customer>
Provided: SdkIterable<Page<Customer>>
```
Which is kind of strange because 
`public interface PageIterable<T> extends SdkIterable<Page<T>>`

I can circumvent this by using `PageIterable.create(index.query())`

I am wondering if this is a documentation issue, a bug or me doing something wrong.
I'd prefer being this a bug, because index.query() should behave like table.query().

Any help is appreciated.

Thanks
Frank

## Links

https://github.com/aws/aws-sdk-java-v2/blob/master/services-custom/dynamodb-enhanced/README.md

**Comment by debora-ito** [comment]
@superwese I can see it too, I will investigate.

Another way to work around it is:
```
Iterator<Page<Customer>> customersWithName =  
        customersByName.query(r -> r.queryConditional(keyEqualTo(k -> k.partitionValue("Smith"))))
        .iterator();
```

**Comment by bigunyak** [comment]
Stumbled upon the same problem.
Definitely looks like a bug, at least in the documentation.

**Comment by RushinNaik** [comment]
+1 for the described issue.

**Comment by pacosw1** [comment]
+1 issues is still not fixed and the docs example does not work.

**Comment by debora-ito** [comment]
Please don't comment with +1s, they are not helpful. Add your upvote to the original issue.

**Comment by debora-ito** [comment]
Documentation was fixed via #2529. Closing.
